packages/salesforce/src/Adaptor.js

```diff
@@ -528,77 +522,40 @@ export function insert(sObjectName, records) {
  * @example <caption>Query patients by Health ID using a lazy state reference</caption>
  * query(`SELECT Id FROM Patient__c WHERE Health_ID__c = '${$.data.healthId}'`);
  * @function
- * @param {(string|function)} query - A SOQL query string or a function that returns a query string. Must be less than 4000 characters in WHERE clause
- * @param {QueryOptions} [options] - Optional configuration for the query operation
+ * @param {string} query - A SOQL query string. Must be less than 4000 characters in WHERE clause
+ * @param {object} [options] - Optional query options, {@link https://jsforce.github.io/jsforce/types/query.QueryOptions.html query options}
  * @state {SalesforceState}
  * @property data - Array of result objects of the form <code>\{ done, totalSize, records \}</code>
  * @returns {Operation}
  */
-export function query(query, options = {}) {
+export function query(query, options) {
   return async state => {
-    const [resolvedQuery, resolvedOptions] = expandReferences(
-      state,
-      query,
-      options
-    );
+    const [
+      resolvedQuery,
+      resolvedOptions = { autoFetch: false, maxFetch: 10000 },
+    ] = expandReferences(state, query, options);
     console.log(`Executing query: ${resolvedQuery}`);
-    const autoFetch = resolvedOptions.autoFetch || resolvedOptions.autofetch;
+    const { autoFetch, maxFetch } = resolvedOptions;
```

Comment:
Is autoFetch actually doing anything now?

Presumably if you pass maxFetch: 10m, it'll just download 10m records regardless of whether autofetch is enabled? This, by the way, is the correct pagination behaviour. Ask for a million records and you should be given a million records.

Reply:
Okay that make sense, but i wonder if we need a better way of querying for more results. Because making the query by default autoPage make sense but you still have limitation of 10000 max records by default, if you need more you just need to adjust `maxFetch` which still is not a guarantee if it will return all results

Reply:
I think `autoFetch` is still useful for when user wants to get the default maximum number of results and get the `nextRecordsUrl` incase they need to use it in their job code

Reply:
But autofetch isn't doing anything right now. So if you explicitly want _all_ records, what do you do?

Also we haven't documented the 10k default? It's probably worth us duplicating documentation on options we think are going to be super common. Like `maxFetch` right now (which in other adaptors we call `limit`, so maybe we should think about that) is basically essential to large queries